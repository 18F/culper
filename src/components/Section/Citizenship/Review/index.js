import Review from './Review'
export default Review
