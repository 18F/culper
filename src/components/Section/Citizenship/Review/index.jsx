import React from 'react'
import PropTypes from 'prop-types'

import { CITIZENSHIP, CITIZENSHIP_REVIEW } from 'config/formSections/citizenship'

import Status from '../Status'
import Multiple from '../Multiple'
import ConnectedPassports from '../Multiple/Passports'

import connectCitizenshipSection from '../CitizenshipConnector'

const sectionConfig = {
  section: CITIZENSHIP.name,
  store: CITIZENSHIP.store,
  subsection: CITIZENSHIP_REVIEW.name,
}

export const Review = ({ requireCitizenshipForeignPassportsSection }) => {
  const subsectionProps = {
    required: true,
    scrollIntoView: false,
  }

  const sectionDivider = (
    <hr className="section-divider" />
  )

  return (
    <div>
      <Status {...subsectionProps} />
      {sectionDivider}
      <Multiple {...subsectionProps} />
      {requireCitizenshipForeignPassportsSection && sectionDivider}
      {requireCitizenshipForeignPassportsSection && <ConnectedPassports {...subsectionProps} />}
    </div>
  )
}

Review.propTypes = {
  requireCitizenshipForeignPassportsSection: PropTypes.bool,
}

Review.defaultProps = {
  requireCitizenshipForeignPassportsSection: true,
}

export default connectCitizenshipSection(Review, sectionConfig)
