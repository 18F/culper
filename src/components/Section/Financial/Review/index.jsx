import React from 'react'
import PropTypes from 'prop-types'

import { FINANCIAL, FINANCIAL_REVIEW } from 'config/formSections/financial'

import Bankruptcies from 'components/Section/Financial/Bankruptcy'
import Gambling from 'components/Section/Financial/Gambling'
import Taxes from 'components/Section/Financial/Taxes'
import Card from 'components/Section/Financial/Card'
import Credit from 'components/Section/Financial/Credit'
import Delinquent from 'components/Section/Financial/Delinquent'
import Nonpayment from 'components/Section/Financial/Nonpayment'

import connectSubsection from 'components/Section/shared/SubsectionConnector'

const sectionConfig = {
  section: FINANCIAL.name,
  store: FINANCIAL.store,
  subsection: FINANCIAL_REVIEW.name,
}

const Review = ({
  requireFinancialBankruptcySection,
  requireFinancialGamblingSection,
  requireFinancialTaxesSection,
  requireFinancialCardSection,
  requireFinancialCreditSection,
  requireFinancialDelinquentSection,
  requireFinancialNonpaymentSection,
}) => {
  const subsectionProps = {
    required: true,
    scrollIntoView: false,
    scrollToBottom: undefined,
  }

  const sectionDivider = (
    <hr className="section-divider" />
  )

  return (
    <div>
      {requireFinancialBankruptcySection && (
        <span>
          <Bankruptcies {...subsectionProps} />
          {sectionDivider}
        </span>
      )}

      {requireFinancialGamblingSection && (
        <span>
          <Gambling {...subsectionProps} />
          {sectionDivider}
        </span>
      )}

      {requireFinancialTaxesSection && (
        <span>
          <Taxes {...subsectionProps} />
          {sectionDivider}
        </span>
      )}

      {requireFinancialCardSection && (
        <span>
          <Card {...subsectionProps} />
          {sectionDivider}
        </span>

      )}

      {requireFinancialCreditSection && (
        <span>
          <Credit {...subsectionProps} />
          {sectionDivider}
        </span>

      )}

      {requireFinancialDelinquentSection && (
        <span>
          <Delinquent {...subsectionProps} />
          {sectionDivider}
        </span>
      )}

      {requireFinancialNonpaymentSection && (
        <Nonpayment {...subsectionProps} />
      )}
    </div>
  )
}

Review.propTypes = {
  requireFinancialBankruptcySection: PropTypes.bool,
  requireFinancialGamblingSection: PropTypes.bool,
  requireFinancialTaxesSection: PropTypes.bool,
  requireFinancialCardSection: PropTypes.bool,
  requireFinancialCreditSection: PropTypes.bool,
  requireFinancialDelinquentSection: PropTypes.bool,
  requireFinancialNonpaymentSection: PropTypes.bool,
}

Review.defaultProps = {
  requireFinancialBankruptcySection: true,
  requireFinancialGamblingSection: true,
  requireFinancialTaxesSection: true,
  requireFinancialCardSection: true,
  requireFinancialCreditSection: true,
  requireFinancialDelinquentSection: true,
  requireFinancialNonpaymentSection: true,
}

export default connectSubsection(Review, sectionConfig)
