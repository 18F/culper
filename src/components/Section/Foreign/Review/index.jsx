import React from 'react'

import { FOREIGN, FOREIGN_REVIEW } from 'config/formSections/foreign'

import Contacts from '../Contacts'
import {
  DirectActivity, IndirectActivity, RealEstateActivity, BenefitActivity, Support,
} from '../Activities'
import {
  Advice, Family, Employment, Ventures, Conferences, Contact, Sponsorship, Political, Voting,
} from '../Business'
import Travel from '../Travel'

import connectForeignSection from '../ForeignConnector'

const sectionConfig = {
  section: FOREIGN.name,
  store: FOREIGN.store,
  subsection: FOREIGN_REVIEW.name,
}

export const Review = ({
  requireForeignContactsSection,
  requireForeignActivitiesSection,
  requireForeignBusinessSection,
  requireForeignTravelSection,
}) => {
  const props = {
    required: true,
    scrollIntoView: false,
  }
  const sectionDivider = (
    <hr className="section-divider" />
  )

  return (
    <div>
      {requireForeignContactsSection && (
        <span>
          <Contacts {...props} />
          {sectionDivider}
        </span>
      )}

      {requireForeignActivitiesSection && (
        <span>
          <DirectActivity {...props} />
          {sectionDivider}
          <IndirectActivity {...props} />
          {sectionDivider}
          <RealEstateActivity {...props} />
          {sectionDivider}
          <BenefitActivity {...props} />
          {sectionDivider}
          <Support {...props} />
          {sectionDivider}
        </span>
      )}

      {requireForeignBusinessSection && (
        <span>
          <Advice {...props} />
          {sectionDivider}
          <Family {...props} />
          {sectionDivider}
          <Employment {...props} />
          {sectionDivider}
          <Ventures {...props} />
          {sectionDivider}
          <Conferences {...props} />
          {sectionDivider}
          <Contact {...props} />
          {sectionDivider}
          <Sponsorship {...props} />
          {sectionDivider}
          <Political {...props} />
          {sectionDivider}
          <Voting {...props} />
          {sectionDivider}
        </span>
      )}

      {requireForeignTravelSection && (
        <Travel {...props} />
      )}
    </div>
  )
}

export default connectForeignSection(Review, sectionConfig)
