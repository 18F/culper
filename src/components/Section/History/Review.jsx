import React from 'react'

import { i18n } from '@config'

import { Show } from '@components/Form'

import connectHistorySection from './HistoryConnector'
import { HISTORY, HISTORY_REVIEW } from '@config/formSections/history'

import ResidenceSummaryProgress from './Residence/ResidenceSummaryProgress'
import EmploymentSummaryProgress from './Employment/EmploymentSummaryProgress'
import EducationSummaryProgress from './Education/EducationSummaryProgress'
import Residence from './Residence'
import Employment from './Employment'
import EducationWrapper from './Education/EducationWrapper'
import FederalWrapper from './Federal/FederalWrapper'

const sectionConfig = {
  section: HISTORY.name,
  store: HISTORY.store,
  subsection: HISTORY_REVIEW.name,
}

const Review = (props) => {
  const { Birthdate } = props

  const subsectionProps = {
    required: true,
    scrollIntoView: false,
    overrideInitial: true,
  }

  const sectionDivider = (
    <hr className="section-divider" />
  )

  const hasAttendedSchool = props.Education.HasAttended.value === 'Yes'
  const hasDegree = props.Education.HasDegree10.value === 'Yes'

  return (
    <div>
      <ResidenceSummaryProgress
        Residence={props.Residence}
        Birthdate={Birthdate} />

      <EmploymentSummaryProgress
        Employment={props.Employment}
        Birthdate={Birthdate} />
      
      <Show when={hasAttendedSchool || hasDegree}>
        <EducationSummaryProgress
          Education={props.Education}
          Birthdate={Birthdate} />
      </Show>

      <hr className="section-divider" />
      <h1 className="section-header">
        {i18n.t('history.residence.collection.caption')}
      </h1>
      <Residence {...subsectionProps} />

      <hr className="section-divider" />
      <h1 className="section-header">
        {i18n.t('history.employment.default.collection.caption')}
      </h1>
      <Employment {...subsectionProps} realtime={true} />

      <hr className="section-divider" />
      <h1 className="section-header">
        {i18n.t('history.education.collection.caption')}
      </h1>
      <EducationWrapper inReview={true} />

      <hr className="section-divider" />
      <FederalWrapper inReview={true} />
    </div>
  )
}

export default connectHistorySection(Review, sectionConfig)
