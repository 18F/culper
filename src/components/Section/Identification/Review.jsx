import React from 'react'

import ApplicantName from './ApplicantName'
import ApplicantBirthDate from './ApplicantBirthDate'
import ApplicantBirthPlace from './ApplicantBirthPlace'

import ApplicantSSN from './ApplicantSSN'
import OtherNames from './OtherNames'
import Physical from './Physical'
import ContactInformation from './ContactInformation'

class Review extends React.Component {
  render () {
    const subsectionProps = {
      required: true,
      scrollIntoView: false,
    }

    return (
      <div>
        <ApplicantName {...subsectionProps} />
        <hr className="section-divider" />
        <ApplicantBirthDate {...subsectionProps} />
        <hr className="section-divider" />
        <ApplicantBirthPlace {...subsectionProps}  />

        {/*  
        <hr className="section-divider" />
        <ApplicantSSN
          name="ssn"
          {...this.props.ApplicantSSN}
          section="identification"
          subsection="ssn"
          dispatch={this.props.dispatch}
          onUpdate={this.props.onUpdate.bind(this, 'ApplicantSSN')}
          onError={this.props.onError}
          required={true}
          scrollIntoView={false}
        />
  
        <hr className="section-divider" />
        <OtherNames
          name="othernames"
          {...this.props.OtherNames}
          section="identification"
          subsection="othernames"
          defaultState={false}
          dispatch={this.props.dispatch}
          onUpdate={this.props.onUpdate.bind(this, 'OtherNames')}
          onError={this.props.onError}
          required={true}
          scrollIntoView={false}
        />
  
        <hr className="section-divider" />
        <ContactInformation
          name="contacts"
          {...this.props.Contacts}
          section="identification"
          subsection="contacts"
          minimumPhoneNumbers={1}
          minimumEmails={1}
          shouldFilterEmptyItems={true}
          defaultState={false}
          dispatch={this.props.dispatch}
          onUpdate={this.props.onUpdate.bind(this, 'Contacts')}
          onError={this.props.onError}
          required={true}
          scrollIntoView={false}
        />
  
        <hr className="section-divider" />
        <Physical
          name="physical"
          {...this.props.Physical}
          section="identification"
          subsection="physical"
          dispatch={this.props.dispatch}
          onUpdate={this.props.onUpdate.bind(this, 'Physical')}
          onError={this.props.onError}
          required={true}
          scrollIntoView={false}
        />*/}
      </div>
    )
  }
}

export default Review
