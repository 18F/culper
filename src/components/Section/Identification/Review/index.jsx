import React from 'react'

import ApplicantName from '../ApplicantName'
import ApplicantBirthDate from '../ApplicantBirthDate'
import ApplicantBirthPlace from '../ApplicantBirthPlace'
import ApplicantSSN from '../ApplicantSSN'
import OtherNames from '../OtherNames'
import ContactInformation from '../ContactInformation'
import Physical from '../Physical'

const Review = () => {
  const subsectionProps = {
    required: true,
    scrollIntoView: false,
  }

  const sectionDivider = (
    <hr className="section-divider" />
  )

  return (
    <div>
      <ApplicantName {...subsectionProps} />
      {sectionDivider}
      <ApplicantBirthDate {...subsectionProps} />
      {sectionDivider}
      <ApplicantBirthPlace {...subsectionProps} />
      {sectionDivider}
      <ApplicantSSN {...subsectionProps} />
      {sectionDivider}
      <OtherNames {...subsectionProps} />
      {sectionDivider}
      <ContactInformation
        {...subsectionProps}
        shouldFilterEmptyItems
      />
      {sectionDivider}
      <Physical {...subsectionProps} />
    </div>
  )
}

export default Review
