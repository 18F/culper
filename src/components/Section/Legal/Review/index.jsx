import React from 'react'
import PropTypes from 'prop-types'

import { LEGAL, LEGAL_REVIEW } from 'config/formSections/legal'

import connectSubsection from 'components/Section/shared/SubsectionConnector'

import ConnectedOffenses from '../Police/Offenses'
import ConnectedOtherOffenses from '../Police/OtherOffenses'
import ConnectedDomesticViolenceList from '../Police/DomesticViolenceList'
import { History, Revoked, Debarred } from '../Investigations'
import ConnectedNonCriminalCourtActions from '../NonCriminalCourtActions'
import { Unauthorized, Manipulating, Unlawful } from '../Technology'
import {
  TerroristOrganization,
  MembershipOverthrow,
  MembershipViolence,
  EngagedInTerrorism,
  Advocating,
  ActivitiesToOverthrow,
  TerrorismAssociation,
} from '../Associations'

const sectionConfig = {
  section: LEGAL.name,
  store: LEGAL.store,
  subsection: LEGAL_REVIEW.name,
}

export const Review = ({
  requireLegalOtherOffensesSection,
  requireLegalNonCriminalCourtSection,
  requireLegalTechnologySection,
}) => {
  const props = {
    required: true,
    scrollIntoView: false,
  }

  const sectionDivider = (
    <hr className="section-divider" />
  )

  return (
    <div>
      <ConnectedOffenses {...props} />

      {requireLegalOtherOffensesSection && (
        <span>
          {sectionDivider}
          <ConnectedOtherOffenses {...props} />
        </span>
      )}

      {sectionDivider}
      <ConnectedDomesticViolenceList {...props} />

      {sectionDivider}
      <History {...props} />
      {sectionDivider}
      <Revoked {...props} />
      {sectionDivider}
      <Debarred {...props} />

      {requireLegalNonCriminalCourtSection && (
        <span>
          {sectionDivider}
          <ConnectedNonCriminalCourtActions {...props} />
        </span>
      )}

      {requireLegalTechnologySection && (
        <span>
          {sectionDivider}
          <Unauthorized {...props} />
          {sectionDivider}
          <Manipulating {...props} />
          {sectionDivider}
          <Unlawful {...props} />
        </span>
      )}

      {sectionDivider}
      <TerroristOrganization {...props} />
      {sectionDivider}
      <EngagedInTerrorism {...props} />
      {sectionDivider}
      <Advocating {...props} />
      {sectionDivider}
      <MembershipOverthrow {...props} />
      {sectionDivider}
      <MembershipViolence {...props} />
      {sectionDivider}
      <ActivitiesToOverthrow {...props} />
      {sectionDivider}
      <TerrorismAssociation {...props} />
    </div>
  )
}

Review.propTypes = {
  requireLegalOtherOffensesSection: PropTypes.bool,
  requireLegalNonCriminalCourtSection: PropTypes.bool,
  requireLegalTechnologySection: PropTypes.bool,
}

Review.defaultProps = {
  requireLegalOtherOffensesSection: true,
  requireLegalNonCriminalCourtSection: true,
  requireLegalTechnologySection: true,
}

export default connectSubsection(Review, sectionConfig)
