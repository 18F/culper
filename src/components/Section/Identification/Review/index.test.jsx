import React from 'react'
import { shallow } from 'enzyme'

import Review from './index'

describe('Identification Review', () => {
  it('renders without errors', () => {
    const component = shallow(<Review />)

    expect(component.exists()).toBe(true)
    expect(component).toMatchSnapshot()
  })
})
