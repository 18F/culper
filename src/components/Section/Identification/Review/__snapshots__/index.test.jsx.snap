// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identification Review renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
    />
    <hr
      className="section-divider"
    />
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
    />
    <hr
      className="section-divider"
    />
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
    />
    <hr
      className="section-divider"
    />
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
    />
    <hr
      className="section-divider"
    />
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
    />
    <hr
      className="section-divider"
    />
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
      shouldFilterEmptyItems={true}
    />
    <hr
      className="section-divider"
    />
    <Connect(ConnectedIdentificationSection)
      required={true}
      scrollIntoView={false}
    />
  </div>,
  "nodes": Array [
    <div>
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
      />
      <hr
        className="section-divider"
      />
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
      />
      <hr
        className="section-divider"
      />
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
      />
      <hr
        className="section-divider"
      />
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
      />
      <hr
        className="section-divider"
      />
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
      />
      <hr
        className="section-divider"
      />
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
        shouldFilterEmptyItems={true}
      />
      <hr
        className="section-divider"
      />
      <Connect(ConnectedIdentificationSection)
        required={true}
        scrollIntoView={false}
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Review />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
            shouldFilterEmptyItems={true}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
            shouldFilterEmptyItems={true}
          />
          <hr
            className="section-divider"
          />
          <Connect(ConnectedIdentificationSection)
            required={true}
            scrollIntoView={false}
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Review />,
}
`;
